### PR TITLE
[ROCM] enable QR_op on ROCM

### PR DESCRIPTION
**Pull request by micmelesse**
This PR enables the QR op on the rocm platform.

### CODE REVIEW

**Comment by deven-amd** [comment]
@cheshire @chsigg gentle ping

**Comment by stevenireeves** [comment]
@cheshire @chsigg any news on this PR?

File: tensorflow/core/util/gpu_solvers.h [hunk 1]
```diff
@@ -267,7 +273,49 @@ class GpuSolver {
   Status Trsm(rocblas_side side, rocblas_fill uplo, rocblas_operation trans,
               rocblas_diagonal diag, int m, int n, const Scalar* alpha,
               const Scalar* A, int lda, Scalar* B, int ldb);
-#else  // GOOGLE_CUDA
```
Comment (cheshire) [inline tensorflow/core/util/gpu_solvers.h:270]: Is it meant to say "ROCM"?

File: tensorflow/core/util/gpu_solvers.h [hunk 2]
```diff
@@ -267,7 +273,49 @@ class GpuSolver {
...
+  // See: http://docs.nvidia.com/cuda/cusolver/#cuds-lt-t-gt-geqrf
+  template <typename Scalar>
+  Status Geqrf(int m, int n, Scalar* dev_A, int lda, Scalar* dev_tau,
+               int* dev_lapack_info);
+
+
+  // This function performs the matrix-matrix addition/transposition
+  //   C = alpha * op(A) + beta * op(B).
+  // Returns Status::OK() if the kernel was launched successfully.  See:
+  // http://docs.nvidia.com/cuda/cublas/index.html#cublas-lt-t-gt-geam
```
Comment (cheshire) [inline tensorflow/core/util/gpu_solvers.h:289]: Was this code moved from elsewhere? If it's ROCm-based, why does it reference nvidia docs?

**Review by cheshire — CHANGES_REQUESTED** [review]

File: tensorflow/core/util/gpu_solvers.h (hunk 2 above)
Comment (stevenireeves) [reply tensorflow/core/util/gpu_solvers.h:289]: @cheshire The function declarations in gpu_solvers is for both sides. Originally, this file was cuda_solvers.h but it was changed a few months ago to gpu_solvers.h to better represent the dual functionality. These were probably artifacts of those changes. I'll remove the references to cuBLAS docs.

Comment (micmelesse) [reply tensorflow/core/util/gpu_solvers.h]: @cheshire @stevenireeves  cleaned up the comments.

**Comment by micmelesse** [comment]
@cheshire @stevenireeves changes implemented.

**Comment by stevenireeves** [comment]
@cheshire can we get another look at this?

**Comment by jayfurmanek** [comment]
@cheshire gentle nudge on this one. thx!

**Review by chsigg — APPROVED** [review]

**Comment by micmelesse** [comment]
@chsigg 

The macos error doesnot seem to have anything to do with our change

``` ERROR: /Volumes/BuildData/tmpfs/tmp/bazel/external/build_bazel_rules_apple/tools/realpath/BUILD:9:14: in _apple_genrule_inner rule @build_bazel_rules_apple//tools/realpath:realpath_genrule:
Traceback (most recent call last):
	File "/Volumes/BuildData/tmpfs/tmp/bazel/external/build_bazel_apple_support/rules/apple_genrule.bzl", line 78, column 28, in _apple_genrule_impl
		apple_support.run_shell(
	File "/Volumes/BuildData/tmpfs/tmp/bazel/external/build_bazel_apple_support/lib/apple_support.bzl", line 339, column 26, in _run_shell
		ctx.actions.run_shell(**_kwargs_for_apple_platform(ctx, **kwargs))
Error in run_shell: 'command' must be of type string. passing a sequence of strings as 'command' is deprecated. To temporarily disable this check, set --incompatible_run_shell_command_string=false.
INFO: Repository remote_java_tools_darwin instantiated at:
```

Samething for tflite tests

```   external/local_config_cc/wrapped_clang '-D_FORTIFY_SOURCE=1' -fstack-protector -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG '-DNS_BLOCK_ASSERTIONS=1' '-std=c++11' -fdebug-compilation-dir . -iquote . -iquote bazel-out/darwin-opt/bin -iquote external/com_google_absl -iquote bazel-out/darwin-opt/bin/external/com_google_absl -iquote external/eigen_archive -iquote bazel-out/darwin-opt/bin/external/eigen_archive -iquote external/nsync -iquote bazel-out/darwin-opt/bin/external/nsync -isystem external/eigen_archive -isystem bazel-out/darwin-opt/bin/external/eigen_archive -isystem external/nsync/public -isystem bazel-out/darwin-opt/bin/external/nsync/public -MD -MF bazel-out/darwin-opt/bin/tensorflow/core/common_runtime/_objs/cost_util/cost_util.d -DEIGEN_MPL2_ONLY '-DEIGEN_MAX_ALIGN_BYTES=64' '-frandom-seed=bazel-out/darwin-opt/bin/tensorflow/core/common_runtime/_objs/cost_util/cost_util.o' -isysroot __BAZEL_XCODE_SDKROOT__ -F__BAZEL_XCODE_SDKROOT__/System/Library/Frameworks -F__BAZEL_XCODE_DEVELOPER_DIR__/Platforms/MacOSX.platform/Developer/Library/Frameworks '-mmacosx-version-min=10.14' -DGRPC_BAZEL_BUILD -w '-std=c++14' -no-canonical-prefixes -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c tensorflow/core/common_runtime/cost_util.cc -o bazel-out/darwin-opt/bin/tensorflow/core/common_runtime/_objs/cost_util/cost_util.o)
Execution platform: @local_execution_config_platform//:platform
clang: error: unknown argument: '-fdebug-compilation-dir'
ERROR: /Volumes/BuildData/tmpfs/src/github/tensorflow/tensorflow/core/lib/strings/BUILD:36:11: Compiling tensorflow/core/lib/strings/ordered_code.cc failed: (Exit 1): wrapped_clang failed: error executing command  ```

**Pull request merged**